rs/consensus/src/consensus/validator.rs

```diff
@@ -956,12 +1035,21 @@ impl Validator {
                 // proposals proceed to be checked normally.
             }
 
+            // Skip block proposals with a disqualified rank. We do this after
+            // checking for a fast-path validation, to avoid getting stuck.
+            if disqualified_ranks
+                .get(proposal.height(), proposal.rank())
+                .is_some()
+            {
+                continue;
+            }
+
             // Skip validation and drop the block if it has a higher rank than a
             // known valid block. Note that this must happen after we first allow
             // "block with notarization" validation (see above). Otherwise we may
             // get stuck when a block maker equivocates.
```

Comment:
Is this comment up-to-date?

Reply:
No. Although that statement does apply to the if-statement before, where we ignore disqualified ranks. So I can move the comment up.